_posts/2022-08-31-#25600.md

```diff
@@ -0,0 +1,75 @@
+---
+layout: pr
+date: 2022-08-31
+title: "Advertise `NODE_REPLACE_BY_FEE` and connect to 1 outbound full-rbf peer if `mempoolfullrbf` sets"
+pr: 25600
+authors: [ariard]
+components: ["p2p"]
+host: mzumsande
+status: upcoming
+commit:
+---
+
+## Notes
+- [BIP 125](https://github.com/bitcoin/bips/blob/master/bip-0125.mediawiki) (Replace-by-fee)
+enables unconfirmed transactions to be replaced by conflicting transactions ("double spends") under certain
+conditions (see the BIP for details). If these conditions are met, nodes will replace the transaction
+in their mempool and relay the replacing transaction to their peers.
+
+- One important feature of BIP 125 is that it is opt-in: Only original transactions that signal
+replaceability (using the `nSequence` field on one of its inputs) are subject to replacement.
+If the original transaction didn't signal replaceability, the replacement transaction
+will be neither accepted nor relayed.
+
+- For L2 protocols, the necessity to opt-in to RBF poses DoS problems, see
+[this mailing list post](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2022-June/020557.html)
+for a summary of research.
+Therefore, it has been suggested to make full-RBF possible (removing the necessity to opt in).
+
+- As a first step, [PR 25353](https://github.com/bitcoin/bitcoin/pull/25353) added an optional
+`-mempoolfullrbf` parameter to Bitcoin Core, which is disabled by default.
+If the node operator enables this, the node will ignore the signaling (but will still adhere to other RBF rules).
+
+- This PR goes one step further by implementing a preferential peering solution for peers that
+have `-mempoolfullrbf` enabled. These peers would now signal full-RBF and make additional
+automatic outbound connections to peers that have also signaled full-RBF.
```

Comment:
Might be worth mentioning there is no increase in overall connections because bandwidth is limited, hence "preferential" peering, not just additional peers "for free."

Reply:
Are you sure? Looking at the code, I think it's the latter, and it's meant to be N_RBF additional outbound connections (which currently doesn't to work as intended, I'll leave a comment in the PR), so it would be 8+2+N_RBF outbound connections.
Of course, this means less inbound slots, but for a non-listening node or a node that is not at the maximum of its inbound connections, it should mean an increase in connections and traffic.
What do you mean with "bandwidth is limited"?

Reply:
Nevermind sorry, resolving. I just meant to say that increasing overall peers would increase bandwidth usage. I haven't looked closely but for some reason I assumed the PR wasn't adding extra outbound slots, oops.